### ISSUE TITLE
FilterExpression listed under PIM

### ISSUE THREAD

**Issue by rwmtse** [blackberry/WebWorks-API-Docs#230]
Find should be a topic of its own and FilterExpression should be under Find.

**Comment by tneil** [comment]
FilterExpression can only be used with PIM and find is just the feature id.  All find methods are on the exiting PIM objects.  This is why we placed the FilterExpression under PIM.

**Comment by rwmtse** [comment]
But blackberry.message.Message also has a find function that uses FilterExpression

**Comment by tneil** [comment]
Yeah.. that's true.. the problem is that there are two FilterExpressions.. One used solely for Phone and another used for Message and PIM.

Would it be too confusing to do the following:
- Find
  - FilterExpression
  - Phone Filter Expression

Or would it be better to do:
- Phone
  - FilterExpression
- Message
  - FilterExpression
- PIM
  - FilterExpression

Where the Message and PIM ones went to the same link...

**Comment by rwmtse** [comment]
Hmmm... I don't like either :P  I suppose the latter one is better than the first. Why is phone filter expression different from the generic one?

**Comment by tneil** [comment]
Oh... long story... but it can be summarized by the fact that we delivered Phone as an add-on extension before we moved the APIs up off the OS.  So there was no way to use the existing FilterExpression with Phone.

We should probably re-factor this so that there is just one FilterExpression

**Comment by rwmtse** [comment]
I vote for refactoring if time permits.

**Comment by tneil** [comment]
With a refactor, we would also need to ensure that old apps using the blackberry.phone.Find.FilterExpression do not break and mark it for future deprecation
